File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    queue = []
    for s in scoville:
        heapq.heappush(queue, s)
    answer = 0
    
    while queue[0] < K:
        if len(queue) < 2:
            return -1
        first = heapq.heappop(queue)
        second = heapq.heappop(queue)
        new_s = (first + second * 2)
        heapq.heappush(queue, new_s)
        answer += 1
        
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> queue = new PriorityQueue<>(
            Comparator.comparingInt(a -> a)
        );
        for(int s : scoville) {
            queue.offer(s);
        }
        
        while(queue.peek() < K) {
            if(queue.size() < 2) {
                return -1;
            }
            Integer first = queue.poll();
            Integer second = queue.poll();
            queue.offer(first + second * 2);
            answer += 1;
        }
        
        return answer;
    }
}